dot/state/service.go

```diff
@@ -79,7 +78,7 @@ func (s *Service) UseMemDB() {
 }
 
 // DB returns the Service's database
-func (s *Service) DB() chaindb.Database {
+func (s *Service) DB() *chaindb.BadgerDB {
```

Comment:
This is left as the badger implementation on purpose, this is about to be changed with #2981